src/main.js

```diff
@@ -35,64 +35,83 @@ const storageApi = {};
 /**
  * Encapsulates authentication flow logic.
  * @param {Object} options Options object.
- * @param {string} options.apiKey The firebase API key
+ * @param {string} options.apiKey The Firebase API key.
  * @param {string} options.redirectUri The redirect URL used by OAuth providers.
  * @param {Array.<ProviderOptions|string>} options.providers Array of arguments that will be passed to the addProvider method.
  */
 export default class Auth {
-	constructor({ name = 'default', apiKey, redirectUri, storage = storageApi } = {}) {
-		if (typeof apiKey !== 'string') throw Error('The argument "apiKey" is required');
-
-		Object.assign(this, {
-			name,
-			apiKey,
-			storage,
-			redirectUri,
-			listeners: []
-		});
-
-		this.storage.get(this.sKey('User')).then(user => {
-			this.setState(JSON.parse(user), false);
-			if (this.user)
-				this.refreshIdToken()
-					.then(() => this.fetchProfile())
-					.catch(e => {
-						if (e.message === 'TOKEN_EXPIRED' || e.message === 'INVALID_ID_TOKEN') return this.signOut();
-						throw e;
-					});
-		});
+	constructor({
```

Comment:
From here down there are many style changes that probably happened because of you editor settings.
Make sure to lint and format the code with eslint and prettier's settings of the project.